Packs/ThreatGrid/Integrations/ThreatGrid/ThreatGrid.py

```diff
@@ -1,1233 +1,1520 @@
+    """
+    md5 = sample_details.get('md5')
+    sha1 = sample_details.get('sha1')
+    sha256 = sample_details.get('sha256')
+
+    command_indicator = Common.File(
+        md5=md5,
+        sha1=sha1,
+        sha256=sha256,
+        name=sample_details.get('filename'),
+        dbot_score=dbot_score,
+    )
+    outputs = {
+        "md5": md5,
+        "sha1": sha1,
+        "sha256": sha256,
     }
-    res = {}
-    # add ioc_key_to_path_dict values to result
-    for k, v in ioc_key_to_path_dict.items():
-        val = demisto.get(ioc, v)
-        if val:
-            res[k] = val
-    # add ioc_data_keys_set values to result.Data
-    res['Data'] = {}
-    for key in ioc_data_keys_set:
-        res['Data'][key] = []
-    if demisto.get(ioc, 'data'):
-        for item in demisto.get(ioc, 'data'):
-            for key in ioc_data_keys_set:
-                if demisto.get(item, key):
-                    res['Data'][key].append(demisto.get(item, key))
-    return res
-
-
-def get_analysis_iocs():
-    """
-    Get data about analysis iocs for a given sample or ioc
-    """
-    sample_id = demisto.getArg('id')
-    ioc = demisto.getArg('ioc')
-    url = SUB_API + 'samples/' + sample_id + '/analysis/iocs'
-    if ioc:
-        url += '/' + ioc
-    params = {'api_key': API_KEY}
-    if demisto.getArg('limit'):
-        params['limit'] = demisto.getArg('limit')
-
-    r = req('GET', url, params=params)
-    iocs = []    # type: ignore
-    dbots = []    # type: ignore
-    items = demisto.get(r.json(), 'data.items')    # type: ignore
-    if not items:
-        append_to_analysis_iocs_arrays(iocs, dbots, demisto.get(r.json(), 'data'))
-    else:
-        for k in items:
-            append_to_analysis_iocs_arrays(iocs, dbots, k)
-    md = tableToMarkdown('ThreatGrid Behavioral Indicators for sample: ' + demisto.getArg('id'), iocs,
-                         ['Title', 'Confidence', 'Severity', 'IOC', 'Tags', 'IOCCategory', 'Data'])
-    md += tableToMarkdown('DBot', dbots, ['Indicator', 'Score', 'Type', 'Vendor'])
-    demisto.results({
-        'Type': entryTypes['note'],
-        'EntryContext': {'ThreatGrid.IOCs': iocs, 'DBotScore': dbots},
-        'HumanReadable': md,
-        'ContentsFormat': formats['json'],
-        'Contents': r.json()
-    })
-
-
-def append_to_analysis_iocs_arrays(iocs, dbots, k):
-    """
-    Helper for appending analysis item to ioc an dbot arrays
-    """
-    if k and k.get('ioc'):
-        iocs.append(ioc_to_readable(k))
-        dbots.append({
-            'Vendor': 'Threat Grid',
-            'Type': 'IOC',
-            'Indicator': k.get('ioc'),
-            'Score': calc_score(k.get('severity'))
-        })
+    readable_output = tableToMarkdown(
+        name=f"ThreatGrid File Reputation for {md5} \n",
+        t=outputs,
+    )
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix="ThreatGrid.File",
+        outputs=outputs,
+        outputs_key_field='file',
```

Comment:
is it `file`? I'd expect one of the hashes.

Reply:
You right. Updated to md5.